src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/LanguageServices/CSharpLanguageServiceHost.cs

```diff
@@ -25,14 +26,11 @@ protected override Guid IntelliSenseProviderGuid
             get { return CSharpIntellisenseProvider; }
         }
 
-        /// <summary>
-        /// Invoked when the UnconfiguredProject is first loaded to initialize language services.
-        /// </summary>
-        [ProjectAutoLoad(ProjectLoadCheckpoint.InitialActiveConfigurationKnown)]
+        [ProjectAutoLoad(ProjectLoadCheckpoint.ProjectFactoryCompleted)]
         [AppliesTo(ProjectCapability.CSharpLanguageService)]
-        private void Initialize()
+        private Task OnProjectFactoryCompleted()
         {
-            var nowait = InitializeAsync();
+            return InitializeAsync();
```

Comment:
Unless I'm fundamentally misunderstanding how async works, I'm not convinced. It doesn't matter if we call ConfigureAwait on InitializeAsync or not, that will not stop a continuation from being marshaled back to the original context, here's a little test app I wrote in WinForms (console doesn't have a sync contexts set):

Running:
``` C#
        private async void button1_Click(object sender, EventArgs e)
        {
            await InitializeAsync();

            Console.WriteLine(this.InvokeRequired ? "button1_Click: Background" : "button1_Click: UI");
        }

        private async Task OnProjectFactoryCompletion()
        {
            await InitializeAsync().ConfigureAwait(false);
        }

        private async Task InitializeAsync()
        {
            await Task.Run(() => { }).ConfigureAwait(false);

            Console.WriteLine(this.InvokeRequired ? "InitializeAsync: Background" : "InitializeAsync: UI");
        }
```

Results in:
```
InitializeAsync: Background
button1_Click: UI
```

Running:

``` C#
        private async void button1_Click(object sender, EventArgs e)
        {
            await InitializeAsync();

            Console.WriteLine(this.InvokeRequired ? "button1_Click: Background" : "button1_Click: UI");
        }

        private Task OnProjectFactoryCompletion()
        {
            return InitializeAsync();
        }

        private async Task InitializeAsync()
        {
            await Task.Run(() => { }).ConfigureAwait(false);

            Console.WriteLine(this.InvokeRequired ? "InitializeAsync: Background" : "InitializeAsync: UI");
        }
```

Results in:
```
InitializeAsync: Background
button1_Click: UI
```

Regardless of whether we use ConfigureAwait in OnProjectFactoryCompletion - we have no influence over whether the caller's continuation gets called on the same or original thread.